### ISSUE TITLE
Tag 0.4.0 not found in repository docker.io/palmstonegames/kube-cert-manager

### ISSUE THREAD

**Issue by ahmetb** [PalmStoneGames/kube-cert-manager#60]
Creating k8s/deployment.yaml gives error:

    Error syncing pod, skipping: failed to "StartContainer" for "kube-cert-manager" with
    ErrImagePull: "Tag 0.5.0 not found in repository docker.io/palmstonegames/kube-cert-manager"

The tag doesn't seem to be there? https://hub.docker.com/r/palmstonegames/kube-cert-manager/tags/

**Comment by nanliu** [comment]
Seems like you have to build your own containers for the latest version. I know there's discussion about moving this project to k8s incubator per #33, but happy to submit either a travis config that build/push to docker hub, or we can take advantage of docker hub build pipeline.

On a somewhat related note, we need git tags for all the versions in this repo. I can do git commit archeology,  but wonder if someone who's used this project longer have a better idea?

**Comment by ahmetb** [comment]
@nanliu interesting. makes me wonder

1. how is everybody using 0.5.0 today? is everybody actually building it themselves? That sounds impossible, as there are no release tags on the project.
2. how did image name land in `deployment.yaml` if it never worked in the first place? that sounds like a guaranteed-to-fail installation experience.

**Comment by nanliu** [comment]
So, based on [buildspec.yml](./buildspec.yml) git history:
0.5.0 aeaa115 (no idea where containers exist)
0.4.0 84bb913 [whereisaaron/kube-cert-manager:0.4.0](https://hub.docker.com/r/whereisaaron/kube-cert-manager/tags/)
Based on [deployment.yml](./k8s/deployment.yml) git history and docker inspect create timestamp of [palmstonegames/kube-cert-manager](https://hub.docker.com/r/palmstonegames/kube-cert-manager/tags/)
0.3.1 c779a05
0.3.0 7a5f64e
0.2.0 0cc027d
0.1.1 7f02012
0.1.0 502faa2

Please double check and tag repo if it's correct.

**Comment by rosskukulinski** [comment]
@ahmetb I have my own version that I built manually including some of the now-merged PRs.  Looking forward to moving back to the upstream version once we work out where the project will end up.

**Comment by ahmetb** [comment]
@rosskukulinski that sounds like an independent discussion. Regardless of where this project ends up, there's no reason to tag releases on GitHub and set up simple Docker Hub builds based on git tags.

**Comment by luna-duclos** [comment]
0.5.0 is the currently in development version, which is why it hasn't been tagged yet, this indeed needs to become much more clear from the documentation and will be looked at soonish

**Comment by ahmetb** [comment]
@luna-duclos perhaps a short term fix would be to fix the deployment.yaml with an image that works out of the box.

**Comment by luna-duclos** [comment]
I've created tags up till 0.4.0, the next version I still consider unstable, hence why it wont' get a tag (yet), I've clarified this in the docs as well.

**Comment by nanliu** [comment]
@luna-duclos, would you mind pushing 0.4.0 release to hub.docker.com?

**Comment by luna-duclos** [comment]
I hadn't noticed it wasn't there, will do.

**Comment by roshan-mr** [comment]
Tag 0.4.0 and 0.5.0 not available in docker hub

**Comment by kyrias** [comment]
@luna-duclos Ping? 0.4.0 is still not available on Docker Hub.

**Comment by saward** [comment]
Yep, still only showing up to 0.3.1: https://hub.docker.com/r/palmstonegames/kube-cert-manager/tags/

**Comment by emilingerslev** [comment]
It's still only `0.3.1` that's available. You could also put `0.5.0` up there as `0.5.0-unstable` or `0.5.0-beta`.

**Comment by alex-dixon** [comment]
We appreciate your work! But this kind of sucks.
```
Failed to pull image "palmstonegames/kube-cert-manager:0.5.0": rpc error: code = 2 desc = Error response from daemon: {"message":"manifest for palmstonegames/kube-cert-manager:0.5.0 not found"}
Error syncing pod
```
Followed instructions in `deployment-guide.md` which references `k8s/deployment.yaml` which references image `palmstonegames/kube-cert-manager:0.5.0`.

**Comment by weitzj** [comment]
In the meantime I pushed it to our dockerhub tagged with the git Sha
https://hub.docker.com/r/iosphere/kube-cert-manager/

`docker pull iosphere/kube-cert-manager:c8443d4`

This is the current 0.5 version  and uses Go 1.9. It still needs some CI setup

**Comment by rochdev** [comment]
Is at least a 0.4.0 Docker Hub tag planned since this version was already released?

**Comment by weitzj** [comment]
I anyone is interested, I have pushed in image to our repository:

`iosphere/kube-cert-manager:5bba617`

This contains the latest master commit `5bba617` with Go 1.9.2 and Alpine 3.7
